### ISSUE TITLE
Cannot add any new integration

### ISSUE THREAD

**Issue by MaxFestersen** [LemmusLemmus/GlosSI-Integration-Playnite#19]
![image](https://user-images.githubusercontent.com/49905988/215600215-38f39c20-362d-4f3b-aaf4-c060c4955c30.png)
I cannot add any new integrations. For any game i add, that has not been added before, this warning will appear. 
"More Steam shortcuts appears to have been modified by GlosSiConfig than intended, the changes have been reverted. Please consider reporting this erroneous behavior."
Clicking OK or cancel will lead to a fail notification.
![image](https://user-images.githubusercontent.com/49905988/215601330-fb80b92b-cd48-474f-bfb4-81d8544dc490.png)
"No games were added as GlosSi Steam shortcuts. This could be due to games already having been added or having the ignored tag."
To be clear, this happens to any game, including games that have not been added and do not have the ignored tag. Except for games i have added shortly after the initial installation, that i had the integration removed for and re-added.

**Comment by MaxFestersen** [comment]
Something new happened. It now removed all integrations, except a few. The few left behind seem to be some i added manually, but are stripped of all information, except for icon. Also the Destenation has been changed to GlosSITarget.exe and the folder to the GlosSI install folder. The games that were not removed are also duplicates of games i have in playnite, but not games i used with GlosSi-integration.

Tldr
Games i only had in steam manually were removed.
Games added though GlosSi-integration were removed.
Games that were added manually and are present in playnite were not removed, but had all their information removed. Note: Note: these games were not added again with GlosSi-Integration.

**Comment by MaxFestersen** [comment]
After removing the entries, I can once again add games through GlosSi-integration.

**Comment by LemmusLemmus** [comment]
That is not good.

The error message you are seeing is a temporary "fix" (if it even is a fix that is, since it is completely untested due to me being unable to replicate this bug reliably) for when GlosSI for some reason modifies several shortcuts at once. This was a problem with an older version of GlosSI, so I guess this error message and "fixing" should be removed by now. Either way, the reason the issue occurred this time is evidently still of interest.

A couple of questions:

1. If I understood you correctly, you have been able to "add integration" before. Do you recall anything in particular having changed since then?

2. Did you by any chance update GlosSI recently? The only way to encounter this issue that I am aware of, is when you add shortcuts and have already added some shortcuts using a specific older version of GlosSI (how shortcuts are added changed in some version of GlosSI, leading to incompatibilities with GlosSI shortcuts added before that IIRC).

3. What version of GlosSI are you currently using?

4. Could you provide the extension logs? In the Playnite installation folder there should be a file called `extensions.log`. The logs should hopefully clear things up regarding this irritating bug.

6. Just for clarification; the shortcuts you mentioned having been added manually, were those shortcuts added via Steam or via GlosSIConfig?

7. Although I have not encountered this problem, I'd imagine that this issue could _potentially_ also occur if a Steam shortcut is added via the extension when there already exists a Steam shortcut with the same name. 
There cannot exist two _GlosSI_ Steam shortcuts with the same name, since the configuration files are stored in the same folder and with the name of the game.
A GlosSI Steam shortcut and a Steam shortcut added via Steam could theoretically have the same name, but I have no idea how well GlosSI handles this scenario.
<sub>Even if two shortcuts having the same name is not the cause of the problem this time, I should really handle this situation somehow. For example by overwriting the existing shortcut or by renaming the new shortcut to something else. Or by GlosSI being updated such that two games can have the same name.</sub>


The "This could be due to games already having been added or having the ignored tag." part of the message is pretty vague and could definitely be more specific or removed.


By the way, if you need to [restore your shortcuts](https://github.com/LemmusLemmus/GlosSI-Integration-Playnite/wiki/Reverting-changes) to how they were before using this extension, the extension copies your shortcut files the first time the extension is used.

**Comment by MaxFestersen** [comment]
1. I added ~400 games from itch in bulk with no issue. Many games were removed manually, as the controller support was needed. Many games were uninstalled. I tried to add some steam games by accident, but I was informed by the integration that I had made the mistake, and the action was canceled.
2. I have never updated GlosSi. 
3. I have version 0.1.1.2
4. No. In my installation folder, the only log present is debug.log - and it only says "[0125/230357.722:ERROR:registration_protocol_win.cc(107)] CreateFile: Den angivne fil blev ikke fundet. (0x2)" - translating from Danish to English: The specified file was not found.
5. The shortcuts i added was added through steam before i even had Playnite installed. I have never added shortcuts through GlosSi.
6. I have duplicates of the same games, it could be possible that the same games were added multiple times.

The issue can probably be replicated by adding entries two steam with blank names that target GlosSITarget.exe. Why GlosSi or GlosSi-integration did this, or how to make them do it i don't know.

**Comment by LemmusLemmus** [comment]
Thanks for the information!

Regarding the `extensions.log` file, I misremembered where it is located. It is only located in the installation folder if you are using a portable version of Playnite, otherwise it can be found in `%appdata%\Playnite`. Since you added ~400 games, I imagine the file could be quite large...

<hr>

> it could be possible that the same games were added multiple times.

It seems like that could be the case here and it should definitely be actively avoided in a future version of the extension.

One reason the same game could be added multiple times would be if two games are named the same or just too similarly: since the game names are used as file names, certain characters cannot be used in the file names (the Steam shortcut will still be able to use the characters). So for example if one game is called "game" and another "game?", their file names would be the same since "?" is an illegal file name character that would be removed, resulting in both games using the same configuration file and that same configuration file being added as a Steam shortcut twice.

<hr>
Another possible reason this error could occur is if the "More Steam shortcuts appears to have been modified by GlosSIConfig than intended" check and restoration are both malfunctioning. I guess the logs will tell.

**Comment by MaxFestersen** [comment]
There seems to be some personal information within the files. So i can't post it on a open forum.

**Comment by LemmusLemmus** [comment]
That's reasonable. Feel free to either send it on Discord (you should be able to find me on the GlosSI discord), or only send the relevant information. The only information that _should_ be relevant is the first (and maybe second) time the issue appears (you can search the file for a line containing `|ERROR|GlosSIIntegration#GlosSITarget:More than one shortcut was changed unintentionally by GlosSIConfig.` If there exists a warning that says `|WARN |GlosSIIntegration#GlosSITarget:No shortcuts were changed by GlosSIConfig.`, it might also be relevant). Feel free to strip any paths of your username or other folders, the paths themselves should not be relevant (with the exception of whether they lead to a manually added game or a GlosSI shortcut). Feel free to hide game/shortcut names as well, as long as you do not hide the name of the game for which the issue occurs (or any game names that are similar). I'm only really interested in what changed in the shortcuts.vdf file and in what situation.
